app/scripts/models/flow.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent a login flow, holding the data that we
+ * need to submit as part of our metrics when we perform actions
+ * during the flow.
+ *
+ * Try to fetch the flowId and flowBegin from the resume token.
+ * If unavailable there, fetch from data attributes in the DOM.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var Backbone = require('backbone');
+  var Cocktail = require('cocktail');
+  var ErrorUtils = require('lib/error-utils');
+  var ResumeTokenMixin = require('models/mixins/resume-token');
+  var SearchParamMixin = require('models/mixins/search-param');
```

Comment:
There is probably also a case for having the resume token mixin fetch the resume token directly from the search parameters itself so consumers don't need it. The consumer could  still pass in a stringified resume token to `populateFromStringifiedResumeToken` to override default behavior, but it wouldn't be required. It feels a little gross passing a window to these models and fetching the data from there, but we discussed that earlier. The grossness has to live somewhere and sometimes it's just easiest to keep it all self-contained.